src/bin/lpython.cpp

```diff
@@ -269,55 +268,84 @@ int emit_llvm(const std::string &infile,
     return 0;
 }
 
+void print_report(std::vector <std::pair<std::string, double>> &times, bool allow) {
+    if (!allow)
+        return;
+    for (std::pair<std::string, double> &stage: times) {
+        std::cout << stage.first << ": " << stage.second << "ms" << std::endl;
+    }
+}
+
 int compile_python_to_object_file(
         const std::string &infile,
         const std::string &outfile,
         const std::string &runtime_library_dir,
-        CompilerOptions &compiler_options)
-{
-    Allocator al(4*1024);
+        const bool time_report,
+        CompilerOptions &compiler_options) {
+
+    Allocator al(4 * 1024);
     LFortran::diag::Diagnostics diagnostics;
     LFortran::LocationManager lm;
+    std::vector <std::pair<std::string, double>> times;
+    
+    auto start_file = std::chrono::high_resolution_clock::now();
     lm.in_filename = infile;
     std::string input = LFortran::read_file(infile);
     lm.init_simple(input);
-    LFortran::Result<LFortran::LPython::AST::ast_t*> r = parse_python_file(
-        al, runtime_library_dir, infile, diagnostics, compiler_options.new_parser);
+    auto end_file = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("Reading_file", std::chrono::duration<double, std::milli>(end_file - start_file).count()));
+
+    auto start_parser = std::chrono::high_resolution_clock::now();
+    LFortran::Result < LFortran::LPython::AST::ast_t * > r = parse_python_file(
+            al, runtime_library_dir, infile, diagnostics, compiler_options.new_parser);
+    auto end_parser = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("parsing", std::chrono::duration<double, std::milli>(end_parser - start_parser).count()));
     std::cerr << diagnostics.render(input, lm, compiler_options);
     if (!r.ok) {
+        print_report(times, time_report);
```

Comment:
I have to test it, it seems it would always print the report if the parsing fails?

Reply:
yes, and even if ast to asr  or asr to llvm, it will print last stage even if it faild.
i thought it will be usefull to know time even if stage faild, but i  will remove it.

Reply:
I think it would be useful even if it fails.

Reply:
ok, thats fine i will bring it back.